### PR TITLE
Start of project

### PR DESCRIPTION
**Pull request by Artsapollo**

### CODE REVIEW

File: src/main/java/springboot/controller/CustomersController.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+@RequestMapping("/customer")
+public class CustomersController {
+
+    private static final Logger LOG = LogManager.getLogger(CustomersController.class);
+
+    @Autowired
+    private CustomersService customersService;
+
+    @GetMapping
+    public @ResponseBody String test(){
```
Comment (Cherkashyn) [inline src/main/java/springboot/controller/CustomersController.java]: Delete test

File: src/main/java/springboot/controller/CustomersController.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    }
+
+    @PutMapping("/{id}")
+        public void updateCustomerById(@PathVariable("id") BigDecimal id, @RequestParam("cust_num") BigDecimal cust_num){
+        LOG.info("Updating Customer, id={}, cust_num={}", id, cust_num);
+        Customer customer = customersService.findCustomerById(id);
+        if(Objects.isNull(customer)){
+            LOG.warn("Can't update not existing customer");
+        } else {
+            customer.setCustNum(cust_num);
```
Comment (Cherkashyn) [inline src/main/java/springboot/controller/CustomersController.java]: I'm not sure updating entity's id a good choice.

File: src/main/java/springboot/controller/CustomersController.java [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    }
+
+    @DeleteMapping("/{id}")
+    public void deleteCustomerById(@PathVariable("id") BigDecimal id){
+        LOG.info("Deleting Customer By Id, id={}", id);
+        customersService.deleteCustomer(id);
+        LOG.info("Customer deleted");
+    }
+
+}
```
Comment (Cherkashyn) [inline src/main/java/springboot/controller/CustomersController.java]: Did you miss getAll method?

File: src/main/java/springboot/dao/CustomersDaoImpl.java [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+        }
+        this.sessionFactoryObj = factory.unwrap(SessionFactory.class);
+    }
+
+    @Override
+    public Set<Customer> getAllCustomers() {
+        Set<Customer> customers = new HashSet<>();
+        try {
+            sessionObj = sessionFactoryObj.openSession();
+            customers = new HashSet<>(sessionObj.createQuery("from Customer", Customer.class).list());
```
Comment (Cherkashyn) [inline src/main/java/springboot/dao/CustomersDaoImpl.java]: Why you didn't use JpaRepository interface?

File: src/main/java/springboot/entity/Customer.java [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+@Table(name = "CUSTOMERS", schema = "MA_STUDENT")
+public class Customer implements java.io.Serializable {
+
+    @Id
+    @Column(name = "CUST_NUM", unique = true, nullable = false, precision = 22, scale = 0)
+    private BigDecimal custNum;
+
+    @ManyToOne(fetch = FetchType.LAZY)
+    @JoinColumn(name = "CUST_REP")
+    private Salesrep salesrep;
```
Comment (Cherkashyn) [inline src/main/java/springboot/entity/Customer.java]: Is it works fine without @JsonIgnore?

File: src/main/resources/log4j2.xml [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+            <Policies>
+                <SizeBasedTriggeringPolicy size="30 MB" />
+            </Policies>
+            <DefaultRolloverStrategy max="2" />
+        </RollingFile>
+
+    </Appenders>
+    <Loggers>
+        <AsyncLogger name="org.hibernate" level="error" />
+        <AsyncLogger name="org.springframework" level="debug" />
```
Comment (SergeyPasko) [inline src/main/resources/log4j2.xml]: level=error

**Pull request merged**